belt-node8-service.gitlab-ci.yml

```diff
@@ -0,0 +1,83 @@
+image: briangweber/node:8
+
+before_script:
+  - echo "@cimpress-technology:registry=https://artifactory.cimpress.io/artifactory/api/npm/npm-release-local/" > ~/.npmrc
+  - echo "//artifactory.cimpress.io/artifactory/api/npm/npm-release-local/:_authToken=${CT_ARTIFACTORY_NPM_TOKEN}" >> ~/.npmrc
+  - npm i -g grunt
+  - npm ci
+  - export VERSION=${CI_COMMIT_SHA:0:7}
+
+stages:
+  - build
+  - validate
+  - publish
+  - deploy
+
+build:
+  stage: build
+  script:
+    - npm test
+  coverage: '/Branches\s*:\s*(\d*\.?\d+)%/'
+  artifacts:
+    paths:
+      - coverage/
+
+dependency_scanning:
+  image: briangweber/docker-node:latest
+  stage: build
+  allow_failure: true
+  services:
+    - docker:stable-dind
+  script:
+    - export SP_VERSION=$(echo "$CI_SERVER_VERSION" | sed 's/^\([0-9]*\)\.\([0-9]*\).*/\1-\2-stable/')
+    - docker run
+        --env DEP_SCAN_DISABLE_REMOTE_CHECKS="${DEP_SCAN_DISABLE_REMOTE_CHECKS:-false}"
+        --volume "$PWD:/code"
+        --volume /var/run/docker.sock:/var/run/docker.sock
+        "registry.gitlab.com/gitlab-org/security-products/dependency-scanning:$SP_VERSION" /code
+  artifacts:
+    paths: [gl-dependency-scanning-report.json]
+
+publish:
+  image: briangweber/docker-node:carbon
+  stage: publish
+  script:
+    - grunt bundle --packageVersion=$VERSION --packageName=$CI_PROJECT_NAME
+  only:
+    - develop
+  except:
+    - schedules
+
+deploy-cit:
+  stage: deploy
+  environment:
+    name: cit
+  script:
+    - grunt deploy --environment=cit --packageVersion=$VERSION --packageName=$CI_PROJECT_NAME
+  only:
+    - develop
+  except:
+    - schedules
+
+deploy-int:
+  stage: deploy
```

Comment:
You could also use a template to reduce the repetition here, like this example in [Order Manager](https://cimpress.githost.io/commerce/b2b/order-manager/blob/master/.gitlab-ci.yml#L65).